packages/near-api-js/src/account.ts

```diff
@@ -629,4 +630,40 @@ export class Account {
             available: availableBalance.toString()
         };
     }
+
+    /**
+     * Returns the NEAR tokens balance of a given account that is delegated to the staking pools that are part of the validators set in the current epoch.
+     * 
+     * NOTE: If the tokens are delegated to a staking pool that is currently on pause or does not have enough tokens to participate in validation, they won't be accounted for.
+     * @returns {Promise<string>}
+     */
+    async getActiveDelegatedStakeBalance(): Promise<string>  {
+        const { current_validators, next_validators, current_proposals } = await this.connection.provider.validators(null);
+        const pools:Set<string> = new Set();
+        [...current_validators, ...next_validators, ...current_proposals]
+            .forEach((validator) => pools.add(validator.account_id));
+
+        const prefix = getValidatorRegExp(this.connection.networkId);
+        const block = await this.connection.provider.block({ finality: 'final' });
+        const blockHash = block.header.hash;
+        const promises = [...pools]
+            .filter((v) => v.indexOf('nfvalidator') === -1 && v.match(prefix))
```

Comment:
I don't think NAJ should be limited to only a few pre defined staking pool factories. IMO we need to remove this filter and change L659 to `Promise.allSettled`. If we remove this filter we can remove hardcoded regexes in [validator_reg_exp.ts](https://github.com/near/near-api-js/pull/964/files#diff-fe4ab4432b326f7d1f048a0c4feeb3b14e09483769a4de68841fb748115c4fc9)

@marcinbodnar would you know why we are excluding `accountId`s that have the string `'nfvalidator'`?

Reply:
this part of code was influenced by [staking.js](https://github.com/near/near-wallet/blob/master/packages/frontend/src/redux/actions/staking.js#L409) from wallet repo.
I tried to ran them without the filter previously, and some validators actually error out during `new Contract` step. (So I assumed that some validators need to be excluded therefore I follow the implementation similar to what has been done on wallet repo)

Reply:
The `allSettled` should allow us to get the correct total delegated stake even if some validators error out. I am not sure that we should limit NAJ to only consider staked amounts delegated to pools only from some predefined staking pool factories. Could we get further opinions here? @MaximusHaximus @janewang

Reply:
@marcinbodnar @frol can we get your opinion on this?

Reply:
Instead of hard-coding the account ids, I would suggest you handle the error during fetching the delegated balance. If there is any error except when `error.name = 'HANDLER_ERROR' && error.cause.name = 'CONTRACT_EXECUTION_ERROR'` (and ideally also check the error message has `FunctionCallError(MethodResolveError(MethodNotFound))` error), we can skip this validator, but otherwise we should handle the error (potentially retry), as there could be some network errors or some other glitches, and it is not correct behavior to silently skip staking pools that we just could not get the answer from due to some glitch.

Reply:
We already have retry behavior for `viewFunction` in the [`exponentialBackoff` in `sendJsonRpc` method](https://github.com/near/near-api-js/blob/master/packages/near-api-js/src/providers/json-rpc-provider.ts#L321) which will retry if `error.type === 'TimeoutError'`. So I think we can just consider errors thrown by the `this.viewFunction` call final and conclusive and use `allSettled` here. My main concern with skipping validators that don't match a regex is adding a dependency to NAJ and ignoring arbitrary staking pools and factories that may be deployed or exist in the future.

Reply:
I don't think it is a good idea to ignore errors especially when it comes to balances. It can backfire if you return that a user has zero NEAR delegated balance when the problem was that there was no internet connection or RPC was down.